lib/apis/v3/felixconfig.go

```diff
@@ -265,6 +265,35 @@ type FelixConfigurationSpec struct {
 	// modes can use XDP. This is not recommended since it doesn't provide better performance than
 	// iptables. [Default: false]
 	GenericXDPEnabled *bool `json:"genericXDPEnabled,omitempty" confignamev1:"GenericXDPEnabled"`
+
+	// BPFEnabled, if enabled Felix will use the BPF dataplane. [Default: false]
+	BPFEnabled *bool `json:"bpfEnabled,omitempty" validate:"omitempty,"`
+	// BPFLogLevel controls the log level of the BPF programs when in BPF dataplane mode.  One of "Off", "Info", or
+	// "Debug". [Default: Off].
+	BPFLogLevel string `json:"bpfLogLevel" validate:"omitempty,bpfLogLevel"`
+	// BPFDataIfacePattern is a regular expression that controls which interfaces Felix should attach BPF programs to
+	// in order to catch traffic to/from the network.  This needs to match the interfaces that calico workload traffic
+	// flows over as well as any interfaces that handle incoming traffic to nodeports and services from outside the
+	// cluster.  It should not match the workload interfaces (usually named cali...).  [Default: ^(en.*|eth.*|tunl0$)]
```

Comment:
Is this Here To Stay, or just a band-aid for the tech preview?  If the latter, should we tag in some way to be clear that this config field might disappear again without further warning?

Does similar thinking apply to other fields here, or maybe to all the BPF fields?

Reply:
> It should not match the workload interfaces (usually named cali...).

Feels like we should achieve this in code (using `InterfacePrefix`) and so not document it here.

Reply:
There are use cases where you'd want something other than "ever interface that is not a workload interface"  For example, if you have a bond or something then you may want to run the BPF program on the `bond` instead of the `eth`  .  Also possible you have interfaces that you just want to exclude because you're doing something special with them.

Agree, we could auto-exclude workload interfaces based on the prefix.  That's not a bad idea.